### ISSUE TITLE
3.2.5 doesn't see false in this return type declaration?

### ISSUE THREAD

**Issue by jaydiablo** [vimeo/psalm#1525]
Just upgraded to 3.2.5 in a project and Psalm complained about an invalid return type that 3.2.4 wasn't complaining about.

Here's a contrived example on getpsalm.org:  https://psalm.dev/r/ed00d6a241

`false` is clearly defined in the first method's return type, so I don't think psalm should be complaining about it no?

**Comment by weirdan** [comment]
Reducing it further (https://psalm.dev/r/a2d6e585c4)
```php
<?php
 /** @return scalar */
 function f() { return false; }
```
should pass, but fails with `FalsableReturn`.
On the other hand,
```php
<?php
 /** @return scalar */
 function f() { return true; }
```
passes just fine (https://psalm.dev/r/6772777a63).

**Comment by jaydiablo** [comment]
Possibly a regression of this issue?

https://github.com/vimeo/psalm/issues/718

**Comment by muglug** [comment]
@jaydiablo not a regression, just a failure on my part
